### PR TITLE
Add SovrynMultiProcessor

### PR DESCRIPTION
**Pull request by gagiuntoli**
## Context

We add a SovrynMultiProcessor to fetch several prices at once.
The current implementation of the SovrynHelper contract supports this but until now we were not taking advantage of it.

## To-do list

- [ ] Added tests
- [ ] Tested on dev/sbx
- [ ] Changelog was updated with current changes
- [ ] Documentation or guides were updated (slab, readme)
- [ ] Version was updated on `package.json` (releases/hotfixes)

## Checklist before merge

- [ ] **there are no errors in logs in any workers**
- [ ] new blocks are being discovered and marked as finalized
- [ ] foreign blocks are being dispatched
- [ ] squash all commits before merging

### CODE REVIEW

File: src/services/FeedProcessor/MultiFeedProcessor.ts [hunk 1]
```diff
@@ -24,13 +26,19 @@ export default class MultiFeedProcessor {
...
+      this.sovrynMultiProcessor.apply(feedFetchers),
     ]);
 
     promisesResults.forEach((result, i) => {
       if (result.status === 'fulfilled') {
         response = mergeArrays(response, result.value);
       } else {
-        const processors = ['cryptoCompareMultiProcessor', 'coingeckoMultiProcessor', 'uniswapV3'];
+        const processors = [
+          'cryptoCompareMultiProcessor',
```
Comment (DZariusz) [inline src/services/FeedProcessor/MultiFeedProcessor.ts:37]: let's add `TODO` to put all this in eg enum and use here something like Object.keys(...)

File: src/services/dexes/sovryn/SovrynPriceFetcher.ts [hunk 2]
```diff
@@ -51,31 +51,43 @@ weBTC-rUSDT:
           amountIdDecimals: 18
 */
 @injectable()
-export class SovrynPriceFetcher implements FeedFetcherInterface {
+export class SovrynPriceFetcher {
```
Comment (DZariusz) [inline src/services/dexes/sovryn/SovrynPriceFetcher.ts:54]: why we need to remove interface?

Comment (DZariusz) [reply src/services/dexes/sovryn/SovrynPriceFetcher.ts:54]: also I think this fetcher is not longer needed, all we need is multifercher.

File: test/services/sovryn/SovrynPriceFetcher.test.ts [hunk 3]
```diff
@@ -106,15 +106,16 @@ describe('SovrynFetcher-IntegrationTests', () => {
...
       amountInDecimals,
     };
 
     const sovrynPriceFetcher = container.get('SovrynPriceFetcher') as SovrynPriceFetcher;
-    const price = await sovrynPriceFetcher.apply(requestPair);
+    const prices = await sovrynPriceFetcher.apply([requestPair1, requestPair1]);
 
-    console.log('price weBTC/rUSDT:', price);
+    console.log('price weBTC/rUSDT:', prices[0]);
+    console.log('price weBTC/rUSDT:', prices[1]);
```
Comment (DZariusz) [inline test/services/sovryn/SovrynPriceFetcher.test.ts:119]: if we not testing anything, let's use `.skip`

File: src/services/FeedProcessor.ts [hunk 4]
```diff
@@ -181,7 +181,7 @@ class FeedProcessor {
    * @return multiInputs will be aggregated by the respective processor to be fetched in one API call
    */
   private separateInputs(uniqueFeedFetcherMap: {[hash: string]: FeedFetcher}) {
-    const multiFetchingInputsNames = ['CryptoComparePrice', 'CoingeckoPrice', 'UniswapV3'];
+    const multiFetchingInputsNames = ['CryptoComparePrice', 'CoingeckoPrice', 'UniswapV3', 'SovrynPriceFetcher'];
```
Comment (matheushahnn) [inline src/services/FeedProcessor.ts:184]: I think you will need to rebase and use the FetcherName enum here

File: src/services/dexes/sovryn/SovrynMultiProcessor.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  async apply(feedFetchers: FeedFetcher[]): Promise<(number | undefined)[]> {
+    const request = this.createRequest(feedFetchers);
+    return await this.sovrynFetcher.apply(request);
+  }
+
+  private createRequest(feedInputs: FeedFetcher[]): PairRequest[] {
+    const request: PairRequest[] = [];
+
+    feedInputs.forEach((fetcher) => {
+      if (!fetcher.name.includes('SovrynPriceFetcher')) return;
```
Comment (matheushahnn) [inline src/services/dexes/sovryn/SovrynMultiProcessor.ts:20]: Same here

File: src/services/FeedProcessor.ts (hunk 4 above)
Comment (gagiuntoli) [reply src/services/FeedProcessor.ts:184]: Thanks for that

**Comment by matheushahnn** [comment]
@gagiuntoli could you solve the conflict before I review again?
Also, do you want to merge in which order those SoverynMultiProcessor PR's? 
[This is another one](https://github.com/umbrella-network/pegasus/pull/368#pullrequestreview-2100497003)

**Comment by gagiuntoli** [comment]
> @gagiuntoli could you solve the conflict before I review again? Also, do you want to merge in which order those SoverynMultiProcessor PR's? [This is another one](https://github.com/umbrella-network/pegasus/pull/368#pullrequestreview-2100497003)

Hi @matheushahnn I couldn't force push after rebasing develop on this branch. This is why I created a new PR with just changing the branch name. I will close this one.

**Pull request closed without merging**